### PR TITLE
Improve chaos logging

### PR DESCRIPTION
**Pull request by Zelldon**
1. In order to log via json layout, which is better visible and processed
by stackdriver, this commits adds a dependency to the zeebe-util module
and uses the StackDriverLayout
2. Use MDC for logging, add as context information:

 * the job type
 * cluster id
 * cluster plan
 * process instance id
3. Furthermore the process logs are redirected to log4j and uses the
correct context

Example output:

```
{"severity":"DEBUG","logging.googleapis.com/sourceLocation":{"function":"invoke","file":"chaosMain.kt","line":159},"message":"+ kubectl wait --for=condition=Ready pod -l app.kubernetes.io/app=zeebe --timeout=900s -n 0ee5902e-1dfa-4805-9eb2-51d8036faaa2-zeebe","serviceContext":{"service":"chaos-worker","version":"development"},"context":{"clusterPlan":"Production - M","threadId":25,"processInstanceId":"4503599627372936","clusterId":"0ee5902e-1dfa-4805-9eb2-51d8036faaa2","jobType":"verify-readiness.sh","threadPriority":5,"loggerName":"io.zeebe.chaos.ChaosWorker","threadName":"pool-3-thread-1"},"timestampSeconds":1621427016,"timestampNanos":690601000}
{"severity":"DEBUG","logging.googleapis.com/sourceLocation":{"function":"invoke","file":"chaosMain.kt","line":159},"message":"error: no matching resources found","serviceContext":{"service":"chaos-worker","version":"development"},"context":{"clusterPlan":"Production - M","threadId":25,"processInstanceId":"4503599627372936","clusterId":"0ee5902e-1dfa-4805-9eb2-51d8036faaa2","jobType":"verify-readiness.sh","threadPriority":5,"loggerName":"io.zeebe.chaos.ChaosWorker","threadName":"pool-3-thread-1"},"timestampSeconds":1621427016,"timestampNanos":691035000}
{"severity":"INFO","logging.googleapis.com/sourceLocation":{"function":"run","file":"chaosMain.kt","line":87},"message":"Received shutdown signal","serviceContext":{"service":"chaos-worker","version":"development"},"context":{"threadId":33,"threadPriority":5,"loggerName":"io.zeebe.chaos.ChaosWorker","threadName":"Close thread"},"timestampSeconds":1621427054,"timestampNanos":479319000}
```

Interesting is the context part:

```
"context":{"clusterPlan":"Production - M","threadId":25,"processInstanceId":"4503599627372936","clusterId":"0ee5902e-1dfa-4805-9eb2-51d8036faaa2","jobType":"verify-readiness.sh","threadPriority":5,"loggerName":"io.zeebe.chaos.ChaosWorker","threadName":"pool-3-thread-1"}
```

which we can use for filtering in stackdriver

### CODE REVIEW

File: core/chaos-workers/chaos-worker/src/main/kotlin/io/zeebe/chaos/chaosMain.kt [hunk 1]
```diff
@@ -250,3 +271,15 @@ internal fun succeeds(
...
 }
+
+internal fun setMDCForJob(job: ActivatedJob) {
+    MDC.put("jobType", job.type)
+
+    val clusterId = job.variablesAsMap["clusterId"]!! as String
+    MDC.put("clusterId", clusterId)
+
+    val clusterPlan = job.variablesAsMap["clusterPlan"]!! as String
+    MDC.put("clusterPlan", clusterPlan)
```
Comment (pihme) [inline core/chaos-workers/chaos-worker/src/main/kotlin/io/zeebe/chaos/chaosMain.kt:282]: Maybe also interesting:
* clusterName
* region/regionUUID
* generation/generationUUID
* businessKey (a.k.a Jenkins Build URL)

(not sure these are the precise names, though)

File: core/chaos-workers/chaos-worker/src/main/kotlin/io/zeebe/chaos/chaosMain.kt [hunk 2]
```diff
@@ -250,3 +271,15 @@ internal fun succeeds(
...
+
+    MDC.put("processInstanceId", job.processInstanceKey.toString())
```
Comment (pihme) [inline core/chaos-workers/chaos-worker/src/main/kotlin/io/zeebe/chaos/chaosMain.kt:284]:
```suggestion
    MDC.put("processInstanceKey", job.processInstanceKey.toString())
```

File: core/chaos-workers/chaos-worker/src/main/resources/log4j2.xml [hunk 3]
```diff
@@ -1,19 +1,40 @@
...
-  <Root level="error">
-    <AppenderRef ref="Console"/>
-  </Root>
-</Loggers>
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="WARN" shutdownHook="disable">
+  <Properties>
+    <Property name="log.path">logs</Property>
+    <Property name="log.stackdriver.serviceName">chaos-worker</Property>
+    <Property name="log.stackdriver.serviceVersion">${env:ZEEBE_LOG_STACKDRIVER_SERVICEVERSION:-}
```
Comment (pihme) [inline core/chaos-workers/chaos-worker/src/main/resources/log4j2.xml:6]: I don't think this is set in the chaos container

**Review by pihme — APPROVED** [review]
Some suggestions. Looking forward to seeing that in action!

File: core/chaos-workers/chaos-worker/src/main/kotlin/io/zeebe/chaos/chaosMain.kt (hunk 1 above)
Comment (Zelldon) [reply core/chaos-workers/chaos-worker/src/main/kotlin/io/zeebe/chaos/chaosMain.kt:282]: nah.. I never search for these kind of stuff :man_shrugging: we can add it later if we really need it (yagni) except clusterName I also thought about it

**Pull request merged**